From: duos-ipc <noreply@corley.it>
Subject: [PATCH 1/3] freertos: compila il task IPC dentro la lib "comm"

Aggiunge i sorgenti di task/ipc/ alla libreria "comm". I file veri stanno nel
repo duos-ipc e sono raggiunti via symlink creato da scripts/install-rtos.sh;
qui serve solo dire a CMake di compilarli.

Perche' dentro "comm" e non come libreria a se': "comm" e' gia' fra le
CVI_TASK_LIBS linkate in cvirtos.elf e ha gia' sulla include path gli header
di mailbox/rtos_cmdqu che il task usa. Una lib nuova richiederebbe di toccare
anche task/CMakeLists.txt e task/main/CMakeLists.txt.

Generata contro v2.0.1.

diff --git a/freertos/cvitek/task/comm/CMakeLists.txt b/freertos/cvitek/task/comm/CMakeLists.txt
index 8890ed220..f473e0ec6 100644
--- a/freertos/cvitek/task/comm/CMakeLists.txt
+++ b/freertos/cvitek/task/comm/CMakeLists.txt
@@ -1,10 +1,18 @@
 file(GLOB _SOURCES "src/${RUN_ARCH}/*.c")
 
+# duos-ipc: il producer verso Linux vive in task/ipc/ (symlink verso il repo
+# duos-ipc, creato da scripts/install-rtos.sh). Entra nella libreria "comm"
+# invece di essere una lib a se': comm e' gia' linkata in cvirtos.elf e ha gia'
+# gli include di mailbox/rtos_cmdqu di cui il task ha bisogno.
+file(GLOB _IPC_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/../ipc/*.c")
+list(APPEND _SOURCES ${_IPC_SOURCES})
+
 if (CONFIG_FAST_IMAGE_TYPE STRGREATER "0")
 add_compile_definitions(FAST_IMAGE_ENABLE)
 endif()
 
 include_directories(include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../ipc)
 include_directories(${CMAKE_INSTALL_INC_PREFIX}/arch)
 include_directories(${CMAKE_INSTALL_INC_PREFIX}/common)
 include_directories(${CMAKE_INSTALL_INC_PREFIX}/kernel)
